cibuildwheel/linux.py

```diff
@@ -29,10 +30,42 @@ def get_python_configurations(build_selector):
         PythonConfiguration(identifier='cp38-manylinux_i686', path='/opt/python/cp38-cp38'),
         PythonConfiguration(identifier='pp27-manylinux_x86_64', path='/opt/python/pp27-pypy_73'),
         PythonConfiguration(identifier='pp36-manylinux_x86_64', path='/opt/python/pp36-pypy36_pp73'),
+        PythonConfiguration(identifier='cp35-manylinux_aarch64', path='/opt/python/cp35-cp35m'),
+        PythonConfiguration(identifier='cp36-manylinux_aarch64', path='/opt/python/cp36-cp36m'),
+        PythonConfiguration(identifier='cp37-manylinux_aarch64', path='/opt/python/cp37-cp37m'),
+        PythonConfiguration(identifier='cp38-manylinux_aarch64', path='/opt/python/cp38-cp38'),
+        PythonConfiguration(identifier='cp35-manylinux_ppc64le', path='/opt/python/cp35-cp35m'),
+        PythonConfiguration(identifier='cp36-manylinux_ppc64le', path='/opt/python/cp36-cp36m'),
+        PythonConfiguration(identifier='cp37-manylinux_ppc64le', path='/opt/python/cp37-cp37m'),
+        PythonConfiguration(identifier='cp38-manylinux_ppc64le', path='/opt/python/cp38-cp38'),
+        PythonConfiguration(identifier='cp35-manylinux_s390x', path='/opt/python/cp35-cp35m'),
+        PythonConfiguration(identifier='cp36-manylinux_s390x', path='/opt/python/cp36-cp36m'),
+        PythonConfiguration(identifier='cp37-manylinux_s390x', path='/opt/python/cp37-cp37m'),
+        PythonConfiguration(identifier='cp38-manylinux_s390x', path='/opt/python/cp38-cp38'),
     ]
-
     # skip builds as required
-    return [c for c in python_configurations if build_selector(c.identifier)]
+    configurations = []
+    for c in python_configurations:
+        if not build_selector(c.identifier):
+            continue
+        pm = platform.machine()
+        if pm == "x86_64":
+            if c.identifier.endswith('x86_64') or c.identifier.endswith('i686'):
+                configurations.append(c)
+        if pm == "i686":
+            if c.identifier.endswith('i686'):
+                configurations.append(c)
+        elif pm == "aarch64":
+            if c.identifier.endswith('aarch64'):
+                configurations.append(c)
+        elif pm == "ppc64le":
+            if c.identifier.endswith('ppc64le'):
+                configurations.append(c)
+        elif pm == "s390x":
+            if c.identifier.endswith('s390x'):
+                configurations.append(c)
```

Comment:
Would it be worth it to move this to a separate function, such that the logic here could just be `return [c for c in python_configurations if matches_platform(c.identifier) and build_selector(c.identifier)]`?

Reply:
Done in: https://github.com/joerick/cibuildwheel/pull/273/commits/549fc8a93e524b27626cf50d3848a45943622c7a